options/m_option.h

```diff
@@ -404,10 +404,13 @@ char *format_file_size(int64_t size);
 // Enable special semantics for some options when parsing the string "help".
 #define M_OPT_HAVE_HELP         (1 << 7)
 
+// This options prints possible values when called as --option=help
+#define M_OPT_HELP              (1 << 8)
```

Comment:
Isn't that exactly the same thing as above?

Reply:
Actually, it's not. `M_OPT_HAVE_HELP` looks like it's particularly for key-value lists to accept `help` in place of `help=`. It wasn't there when I first wrote this patch, or I'd have named mine differently. The options it applies to aren't parsable by this completion script so they shouldn't have the new option.

The option I'm adding here describes the *output* of `=help`, namely that it conforms to a rough format that the completion script can parse. I can rename it to something like `M_OPT_PARSABLE_HELP` and clarify the comment.

Reply:
I can't say I like this. This stuff is not really mpv's concern. I think there are already too many funny flags that were added only for the zsh script, and have no purpose otherwise.

Reply:
That's fair. It's a little messy in the --list-options output, too. I don't like it either but the ad-hoc nature of mpv's options system doesn't really provide for anything else that I can think of. Though AFAIK the only other flag added for zsh is M_OPT_FILE -- if there are any others I don't think we're using them and they could be removed.

I did actually start a patch for a programmatically usable option dump, which probably just about everyone thinks is a better idea and could be useful outside of zsh completion. Kinda lost interest since nothing was getting merged here, so I didn't get to the hard part of working all the existing options into some common interface for providing valid values. That would have still kept M_OPT_FILE, but could have at least removed it from the --list-options output and wouldn't have added the new flag. I won't make any promises about getting back to working on that soon, though.

It sounds like this PR's approach is fundamentally not acceptable to you, so maybe you should just reject it (or just merge the first commit, which applies to the old code just as well). But if there's anything I can change that will make you happier with this, let me know. I'll keep using it myself regardless, as it's working in practice and doesn't get in the way of much from my perspective. It's been working well for me with only the occasional rebase.

Reply:
> but the ad-hoc nature of mpv's options system

mpv must have one of the most sophisticated option parser systems across CLI programs. Which is actually bad, because it adds complexity.

> I did actually start a patch for a programmatically usable option dump, which probably just about everyone thinks is a better idea and could be useful outside of zsh completion. 

Sounds like it. There's already considerably metadata in option-info (not everything, but improving it would at least benefit much more users than a zsh autocomplete script).

> It sounds like this PR's approach is fundamentally not acceptable to you, so maybe you should just reject it (or just merge the first commit, which applies to the old code just as well).

I can merge the first and last commit (as listed by github), if that's OK with you.